src/js/controller/model.js

```diff
@@ -102,7 +102,7 @@ define([
 
                     /* falls through */
                 case events.JWPLAYER_MEDIA_META:
-                    var duration = data.duration;
+                    var duration = data.duration || 0;
```

Comment:
@hghazzi - @pajong is right that this would now update the model to 0 in cases where the meta event has an invalid duration. The problem here was that `_.isNumber` evaluates to true for `NaN`.